src/theory/sets/theory_sets_rels.cpp

```diff
@@ -1153,8 +1154,9 @@ void TheorySetsRels::check(Theory::Effort level)
       }
       return equal;
     } else if(!a.getType().isBoolean()){
-      makeSharedTerm(a);
-      makeSharedTerm(b);
+      // TODO(project##230): Find a safe type for the singleton operator
```

Comment:
Here I am not certain using `a.getType()` and `b.getType()`  below is safe. It seems this requiring refactoring `TheorySetsRels::areEqual( Node a, Node b )` to have the types for a and b.

Reply:
I dont believe this is critical (the type shouldnt matter here).

But regardless, this requires a lot of refactoring :)